tests/extension/oneapi_bfloat16/bfloat16_api.cpp

```diff
@@ -0,0 +1,225 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+using bfloat16 = sycl::ext::oneapi::bfloat16;
+
+TEST_CASE("Common interface members", "[bfloat16]") {
+  SECTION("default constructor/destructor and copy") {
+    CHECK(std::is_default_constructible_v<bfloat16>);
+    CHECK(std::is_copy_constructible_v<bfloat16>);
+    CHECK(std::is_destructible_v<bfloat16>);
+    CHECK(std::is_copy_assignable_v<bfloat16>);
+  }
+
+  SECTION("float/sycl::half constructors and assignment operators") {
+    float f = 1.f;
+    bfloat16 bf1_float(f);
+    bfloat16 bf2_float = 0;
+    bf2_float = f;
+    {
+      INFO("Check float");
+      CHECK(bf1_float == f);
+      CHECK(bf2_float == f);
+    }
+
+    sycl::half hf = 1.0;
+    bfloat16 bf1_half(hf);
+    bfloat16 bf2_half = 0;
+    bf2_half = hf;
+    {
+      INFO("Check sycl::half");
+      CHECK(bf1_half == hf);
+      CHECK(bf2_half == hf);
+    }
+  }
+
+  SECTION("Conversion") {
+    CHECK(std::is_convertible_v<bfloat16, float>);
+    CHECK(std::is_convertible_v<bfloat16, sycl::half>);
+    CHECK(std::is_convertible_v<bfloat16, bool>);
+    CHECK(std::is_convertible_v<float, bfloat16>);
+    CHECK(std::is_convertible_v<sycl::half, bfloat16>);
+  }
+}
+
+TEST_CASE("Special values", "[bfloat16]") {
+  SECTION("Check that bfloat16 occupies 16 bits of memomry") {
```

Comment:
```suggestion
  SECTION("Check that bfloat16 occupies 16 bits of memory") {
```
Configure your spell-checker.